src/transforms/javascript.rs

```diff
@@ -0,0 +1,789 @@
+use super::Transform;
+use crate::{
+    event::{Event, ValueKind},
+    topology::config::{DataType, TransformConfig},
+};
+use lazy_static::lazy_static;
+use quick_js::{Context, JsValue};
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{HashMap, HashSet},
+    fs,
+    sync::mpsc,
+    thread,
+};
+
+lazy_static! {
+    // although JavaScript identifiers can also contain Unicode characters, we don't allow them
+    static ref JS_VALID_IDENTIFIER: Regex = Regex::new(r"^[a-zA-Z_$][a-zA-Z0-9_$]*$").unwrap();
+    static ref JS_RESERVED_KEYWORDS: HashSet<&'static str> = [
+        // reserved keywords
+        "break",
+        "case",
+        "catch",
+        "class",
+        "const",
+        "continue",
+        "debugger",
+        "default",
+        "delete",
+        "do",
+        "else",
+        "export",
+        "extends",
+        "finally",
+        "for",
+        "function",
+        "if",
+        "import",
+        "in",
+        "instanceof",
+        "new",
+        "return",
+        "super",
+        "switch",
+        "this",
+        "throw",
+        "try",
+        "typeof",
+        "var",
+        "void",
+        "while",
+        "with",
+        "yield",
+        // future reserved keyword
+        "enum",
+        // literals that cannot be used as identifiers
+        "null",
+        "true",
+        "false"
+    ]
+    .iter()
+    .cloned()
```

Comment:
I was thinking about someone exporting a handler with reserved name in a slightly unusual way, for example

```js
this['delete'] = event => {...event, deleted_field: null}
```

and then getting an arcane error `SyntaxError: unexpected token in expression: ''`. Check for reserved keyword would show error `Handler name "delete" is reserved in JavaScript and cannot be used` instead.

Reply:
Ah, I've took this `cloned` in the last example [here](https://doc.rust-lang.org/std/collections/struct.HashMap.html#examples).

Without it, even with `into_iter` instead of `Iter`, the type of the keys is `&&str` instead of `&str`, so that there is a type incompatibility.